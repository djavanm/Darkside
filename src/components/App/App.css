@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

* {
  margin: 0;
}

main {
  display: grid;
  grid-template-columns: 1fr 4fr;
	grid-template-rows: 1fr 5fr;
  grid-template-areas:
    "nav header"
    "nav output";
  height: 100vh;
  background-image: url('http://eskipaper.com/images/star-background-1.jpg');
}

.App-header {
  grid-area: header;
  display: flex;
  justify-content: center;
  align-items: center;
}

h1 {
  font-size: 85px;
  font-family: 'STARWARS';
  color: yellow;
}

h1:hover {
  transform: scale(1.2) perspective(100px);
  transition: all 0.5s ease;
}
