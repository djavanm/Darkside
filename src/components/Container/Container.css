output {
  grid-area: output;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
  padding: 5px;
}

.p_add-favorite {
  font-size: 42px;
  font-family: 'STARWARS';
  color: #ccc7c8;
}

.link {
  text-decoration: none;
  color: black;
}


