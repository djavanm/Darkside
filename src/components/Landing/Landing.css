.Landing_crawl {
  width: 100%;
  background-color: lightblue;
}

.Landing_h2 {
  font-size: 120px;
  margin-bottom: 50px;
  text-align: center;
  text-transform: uppercase;
}

.Landing_p {
  font-size: 60px;
}

.crawler {
  background: transparent !important;
}

.crawler div {
  background: transparent !important;
  min-height: 40vh !important;
}