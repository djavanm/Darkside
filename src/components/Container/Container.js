import React from 'react';
import Card from '../Card/Card'
import './Container.css';

const Container = ({ data, type, toggleFavorite, favorites }) => {
  const dataCards = data.map((card, index) => {
     return <Card className='Card-section 'data={card} type={type} key={index} toggleFavorite={toggleFavorite} favorites={favorites} />
  })
  return (
    <output>
      {type === 'favorites' && !favorites.length && <p className="p_add-favorite">Click a card to favorite</p>}
      {dataCards}
    </output>
  )
}

export default Container;