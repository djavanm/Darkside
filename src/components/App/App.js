import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import Container from '../Container/Container';
import Landing from '../Landing/Landing';
import CardDetails from '../CardDetails/CardDetails';
import apiCalls from '../Util/apiCalls';
import { Route, Link } from 'react-router-dom';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      planets: [],
      people: [],
      vehicles: [],
      favorites: [],
      crawl: {},
      isLoading: true
    }
  }

  toggleFavorite = (card) => {
    const { favorites } = this.state;
    favorites.map(favorite => favorite.name).includes(card.name) ? this.removeFavorite(card) : this.addFavorite(card);
  }

  addFavorite = (card) => {
    const { favorites } = this.state;
    this.setState({ favorites: [...favorites, card]});
  }

  removeFavorite = (card) => {
    const { favorites } = this.state;
    const filteredFavorites = favorites.filter(favorite => favorite.name !== card.name)
    this.setState({ favorites: filteredFavorites});

  }

  componentDidMount() {
    let randomNumber = Math.floor((Math.random() * 7) + 1)
    fetch(`https://swapi.co/api/films/${randomNumber}`)
      .then(res => res.json())
      .then(data => this.setState({ crawl: { title: data.title, body: data.opening_crawl } }))
  

    fetch('https://swapi.co/api/planets/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPlanetData(data.results))
      .then(planets => this.setState({ planets, isLoading: false }))

    fetch('https://swapi.co/api/people/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPeopleData(data.results))
      .then(data => apiCalls.getPeopleHomes(data))
      .then(data => apiCalls.getSpecies(data))
      .then(people => this.setState({ people }))

    fetch('https://swapi.co/api/vehicles/')
      .then(res => res.json())
      .then(data => apiCalls.cleanVehicles(data.results))
      .then(vehicles => this.setState({ vehicles }))
  
    }
    
    render() {
      console.log(this.state)
      const { planets, people, vehicles, favorites, crawl, isLoading } = this.state;
      return (
        <main className="App-main">
        <header className="App-header">
          <Link to='/' className="link">
            <h1>LightSide</h1>
          </Link> 
        </header>
        <Nav favorites={favorites}/>
        <Route exact path='/' render={() => <Landing film={crawl}/> }/>
        <Route exact path='/planets' render={() => <Container data={planets} type="planets" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/people' render={() => <Container data={people} type="people" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/vehicles' render={() => <Container data={vehicles} type="vehicles" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/favorites' render={() => <Container data={favorites} type="favorites" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route path='/planets/:name' render={({ match }) => {
            let targetPlanet = this.state.planets.find(planet => planet.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPlanet} type="planets" />
        }} />
        <Route path='/people/:name' render={({ match }) => {
            let targetPerson = this.state.people.find(person => person.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPerson} type="people" favorites={favorites}/>
        }} />
        <Route path='/vehicles/:name' render={({ match }) => {
            let targetVehicle = this.state.vehicles.find(vehicle => vehicle.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetVehicle} type="vehicles" favorites={favorites}/>
        }} />
        <Route path='/favorites/:name' render={({ match }) => {
            let targetFavorite = this.state.favorites.find(favorite => favorite.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetFavorite} type="favorites" favorites={favorites}/>
        }} />
      </main>
    );
  }
}

export default App;
