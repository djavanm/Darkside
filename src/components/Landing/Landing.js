import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css'
import './Landing.css';
import PropTypes from 'prop-types';

const Landing = ({ film }) => {
  return (
    <Crawl className="Landing_crawl">
      <h2 className="Landing_h2">{film.title}</h2>
      <p className="Landing_p">{film.body}</p>
    </Crawl>
  );
}

Landing.propTypes = {
  film: PropTypes.object.isRequired,
}
export default Landing;