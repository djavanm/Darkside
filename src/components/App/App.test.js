import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import Landing from '../Landing/Landing';
import Container from '../Container/Container';
import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom';

describe('App', () => {
  it.skip('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when toggleFavorite is called once', () => {
    const wrapper = shallow(<App />);
    const mockFavorite = {
      name: 'Djavan'
    };
    const expected = [mockFavorite];

    wrapper.instance().toggleFavorite(mockFavorite);
    expect(wrapper.state('favorites')).toEqual(expected)
  });

  it('should update state when toggleFavorite is called twice', () => {
    const wrapper = shallow(<App />);
    const mockFavorite = {
      name: 'Djavan'
    };
    const expected = [mockFavorite];

    wrapper.instance().toggleFavorite(mockFavorite);
    expect(wrapper.state('favorites')).toEqual([])
    wrapper.instance().toggleFavorite(mockFavorite);
    expect(wrapper.state('favorites')).toEqual(expected)
  });
  
  describe('Route', () => {
    it('should show Landing component for / router', () => {
      const component = mount(<MemoryRouter initialEntries={['/']} >
        <App />
      </MemoryRouter>
      );
      expect(component.find(Landing)).toHaveLength(1);
    });
  
    it('should show Container component for /planets router)', () => {
      const component = mount(<MemoryRouter initialEntries={['/planets']} >
        <App />
      </MemoryRouter>
      );
      expect(component.find(Container)).toHaveLength(1);
    })
  
    it('should show Container component for /people router', () => {
      const component = mount(<MemoryRouter initialEntries={['/people']} >
        <App />
      </MemoryRouter>
      );
      expect(component.find(Container)).toHaveLength(1);
    })
  
    it('should show Container component for /vehicles router', () => {
      const component = mount(<MemoryRouter initialEntries={['/vehicles']} >
        <App />
      </MemoryRouter>
      );
      expect(component.find(Container)).toHaveLength(1);
    });
  });
})
