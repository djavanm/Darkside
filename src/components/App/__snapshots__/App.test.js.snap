// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="App-main null"
>
  <header
    className="App-header"
  >
    <Link
      className="link"
      to="/"
    >
      <h1>
        LightSide
      </h1>
    </Link>
    <img
      alt="Dueling Lightsabers theme picker"
      className="light-sabers"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Dueling_lightsabers.svg/750px-Dueling_lightsabers.svg.png"
    />
  </header>
  <Nav
    favorites={Array []}
    theme="light"
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/planets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/people"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vehicles"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    path="/planets/:name"
    render={[Function]}
  />
  <Route
    path="/people/:name"
    render={[Function]}
  />
  <Route
    path="/vehicles/:name"
    render={[Function]}
  />
  <Route
    path="/favorites/:name"
    render={[Function]}
  />
</main>
`;
